master/internal/core.go

```diff
@@ -411,16 +411,21 @@ func (m *Master) Run() error {
 		groupPath := strings.TrimPrefix(c.Request().URL.Path, webuiBaseRoute+"/")
 		requestedFile := filepath.Join(reactRoot, groupPath)
 		// We do a simple check against directory traversal attacks.
-		// Alternative: https://github.com/cyphar/filepath-securejoin
 		requestedFileAbs, err := filepath.Abs(requestedFile)
 		if err != nil {
 			log.WithError(err).Error("failed to get absolute path to requested file")
 			return c.File(reactIndex)
 		}
-
 		isInReactDir := strings.HasPrefix(requestedFileAbs, reactRootAbs)
+		if !isInReactDir {
+			return echo.NewHTTPError(http.StatusForbidden)
+		}
+		hasMatchingFile, err := fileExists(requestedFile)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "failed to check if file exists")
```

Comment:
Now this returns a 500 if the file didn't exist. Is that ok? Or should we just inline the func and return 400 instead in that case?

Reply:
hmm would it? if the file doesn't exist, it's a directory, fileExists should return `false, nil`
depends on what these errors could be. we don't want leak information about the directory structure and what's available more than we need to.

Reply:
I was going toward making the return value of fileExists more tailored toward its single usecase here and at that point inlining it makes more sense as you mentioned.

Reply:
> we don't want leak information about the directory structure 

although I realize we're open source so that's not as much of a concern